mesonbuild/backend/xcodebackend.py

```diff
@@ -751,6 +753,8 @@ def generate_xc_build_configuration(self):
                     args = pargs + gargs + targs
                     if args:
                         langargs[langnamemap[lang]] = args
+                cargs = list(dict.fromkeys(cargs)) # Remove duplicates, and preserve order
```

Comment:
`dict.fromkeys()` isn't guaranteed to preserve order in python3.6, please use `mesonlib.OrderedSet` for this, it's also what we do elsewhere.

Also, is this actually safe? Sometimes arguments can be given multiple times, but actually matter, like inside a `--start-group/--end-group`.

Reply:
thanks, you're right. Updated.  sorry for the delay